### PR TITLE
kv: avoid O(stores) work in updatePausedFollowersLocked

### LINKED ISSUE #125308: kvserver: do not iterate over all stores in the cluster in `updatePausedFollowersLocked`
While running scalability experiments for the store liveness prototype, we noticed that in the CPU profile, a large percentage of time is spent in `updatePausedFollowersLocked`. 

Screenshot 2024-06-06 at 3 01 20 PM

The experiment (https://github.com/cockroachdb/cockroach/issues/125057) sets up a 100-node cluster with 8 stores on each node, and ~300k ranges. It seems like `updatePausedFollowersLocked` calls `ioThresholdMap.NumAbovePauseThreshold()`, which [iterates](https://github.com/cockroachdb/cockroach/blob/ede2a1bd6db6aa1d4da92c3832ff97a4324bf910/pkg/kv/kvserver/replica_raft_overload.go#L303) over all stores in the cluster, causing unnecessary overhead in multi-store clusters. 

Maybe the behavior here can be improved to iterate only over the stores where there are replicas for the given range, similarly to the logic in `computeExpendableOverloadedFollowers`.

Jira issue: CRDB-39378

### PR DESCRIPTION
**Pull request by nvb**
Fixes #125308.

This commit avoids O(stores) work in Replica.updatePausedFollowersLocked, which is called on every tick.

Release note: None

### CODE REVIEW

**Review by miraradeva — APPROVED** [review]
:lgtm: 

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/125310#-:-NznDCz62YAGfguNSWF1:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten)

File: pkg/kv/kvserver/replica_raft_overload.go [hunk 1]
```diff
@@ -299,7 +299,7 @@ func (osm *ioThresholds) Replace(
 func (r *Replica) updatePausedFollowersLocked(ctx context.Context, ioThresholdMap *ioThresholdMap) {
 	r.mu.pausedFollowers = nil
 
-	if ioThresholdMap.NumAbovePauseThreshold() == 0 {
+	if !ioThresholdMap.AnyAbovePauseThreshold(r.descRLocked().Replicas()) {
```
Comment (pav-kv) [inline pkg/kv/kvserver/replica_raft_overload.go:302]: optional: There are a bunch of `Replicas()` calls below. Could be good to put this in a variable, to make it explicit that all the computation here is based on a list of replicas.

**Review by pav-kv — APPROVED** [review]

File: pkg/kv/kvserver/replica_raft_overload.go (hunk 1 above)
Comment (nvanbenschoten) [reply pkg/kv/kvserver/replica_raft_overload.go]: Done.

**Review by pav-kv — APPROVED** [review]

**Comment by nvanbenschoten** [comment]
TFTRs!

bors r+

**Pull request merged**